trillian/integration/hammer.go

```diff
@@ -1087,12 +1091,28 @@ func HammerCTLog(cfg HammerConfig) error {
 		glog.Info(s.String())
 	})
 
-	for count := uint64(1); count < cfg.Operations; count++ {
-		if err := s.retryOneOp(ctx); err != nil {
-			return err
+	if cfg.TestDuration == 0 {
+		// Test termination is controlled by operation count.
+		for count := uint64(1); count < cfg.Operations; count++ {
+			if err := s.retryOneOp(ctx); err != nil {
+				return err
+			}
+		}
+		glog.Infof("%s: completed %d operations on log", cfg.LogCfg.Prefix, cfg.Operations)
+	} else {
+		// Test termination is controlled by a timeout. Use a separate context
+		// to avoid the last operation propagating an error when the context
+		// times out.
+		ctx2, cancel := context.WithTimeout(context.Background(), cfg.TestDuration)
```

Comment:
Should this be a child of `ctx` so that it can get cancelled if caller wants to quit early?

Reply:
If I do that the operation in flight when it times out gets interrupted so I decided to let it complete. I can change it but will have to check for cancelled errors etc.

Reply:
I've had a look at this and it complicates error handling as the error metrics are incremented before the caller gets to check it. I don't want to suppress `DeadlineExceeded` for everything really.